src/file_traversal.rs

```diff
@@ -6,18 +6,33 @@ use std::fs;
 use walkdir::WalkDir;
 
 pub fn find(config: &Config, result: &mut Vec<MarkupFile>) {
-    let root = &config.directory;
+    let mut relative_root = config.directory.to_owned();
+
+    if let Some(root) = &config.optional.root_dir {
+        relative_root = root
+            .join(&relative_root)
+            .canonicalize()
+            .map_err(|e| {
+                eprintln!(
+                    "Relative path from privided root {:?} is malformed: {:?} --- {}",
+                    root, relative_root, e
+                );
+                std::process::exit(1);
```

Comment:
wooooo I have never seen this: the stdout printed is ***randomly*** truncated from right to left in parts, yielding only some of the message cutoff at the inserted variables or the text between.

Reply:
Resolved _mostly_ using https://doc.rust-lang.org/core/result/enum.Result.html#method.unwrap_or_else

This can be randomly injected into other log lines being printed still. Like the header printout with mlc version